File: CoSim/frontend/src/api/debug.ts
import { authorizedClient } from './client';

export interface DebugStartRequest {
  file_path: string;
  args?: string[];
  port?: number;
}

export interface DebugSessionInfo {
  debug_id: string;
  language?: 'python';
  port: number;
  command: string[];
  working_dir: string;
}

export const startDebugSession = async (
  token: string,
  sessionId: string,
  payload: DebugStartRequest
): Promise<DebugSessionInfo> => {
  const { data } = await authorizedClient(token).post<DebugSessionInfo>(
    `/v1/sessions/${sessionId}/debug/start`,
    payload
  );
  return data;
};

export const stopDebugSession = async (
  token: string,
  sessionId: string,
  debugId: string
): Promise<{ status: string }> => {
  const { data } = await authorizedClient(token).post<{ status: string }>(
    `/v1/sessions/${sessionId}/debug/${debugId}/stop`
  );
  return data;
};

File: CoSim/backend/src/co_sim/schemas/debug.py
from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel, Field


class DebugStartRequest(BaseModel):
    file_path: str = Field(..., min_length=1, max_length=512)
    args: list[str] = Field(default_factory=list)
    port: Optional[int] = Field(default=None, ge=1, le=65535)
    # Reserved for future runtimes; only "python" is currently supported.
    language: Literal["python"] = "python"


class DebugSessionInfo(BaseModel):
    debug_id: str
    language: Literal["python"] = "python"
    port: int
    command: list[str]
    working_dir: str


class DebugStopResponse(BaseModel):
    status: str
